authority/provisioner/k8sSA.go

```diff
@@ -0,0 +1,245 @@
+package provisioner
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/crypto/pemutil"
+	"github.com/smallstep/cli/jose"
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// K8sSAName is the default name used for kubernetes service account tokens,
+	// because there can only be one per instance of step-ca.
+	K8sSAName   = "k8sSA-default"
```

Comment:
I don't see anything restricting this, one k8sSA provisioner per step-ca. Is this a restriction because there's no way to distinguish tokens from different?

If we can only have one, we should do some kind of validation.

Reply:
yep, we can only have one (for now), because the k8s Service Account tokens don't have enough info in them to easily differentiate between kubernetes clusters. The cli restricts the name of the provisioner by always using this name. I can do some kind of validation in the init.